cpp/test/prims/knn_regression.cu

```diff
@@ -0,0 +1,153 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cuda_utils.h>
+#include <gtest/gtest.h>
+#include <test_utils.h>
+#include <iostream>
+#include <vector>
+#include "label/classlabels.h"
+#include "random/rng.h"
+#include "selection/knn.h"
+
+#include "linalg/reduce.h"
+
+//#include <thrust/count.h>
+#include <thrust/device_ptr.h>
+#include <thrust/extrema.h>
+//#include <thrust/reduce.h>
+//#include <thrust/scan.h>
+//#include <thrust/system/cuda/execution_policy.h>
+
+namespace MLCommon {
+namespace Selection {
+
+struct KNNRegressionInputs {
+  int rows;
+  int cols;
+  int n_labels;
+  float cluster_std;
+  int k;
+};
+
+void generate_data(float *out_samples, float *out_labels, int n_rows,
+                   int n_cols, cudaStream_t stream) {
+  Random::Rng r(0ULL, MLCommon::Random::GenTaps);
+
+  r.uniform(out_samples, n_rows * n_cols, 0.0f, 1.0f, stream);
+
+  MLCommon::LinAlg::unaryOp<float>(
+    out_samples, out_samples, n_rows,
+    [=] __device__(float input) { return 2 * input - 1; }, stream);
+
+  MLCommon::LinAlg::reduce(
+    out_labels, out_samples, n_cols, n_rows, 0.0f, true, true, stream, false,
+    [=] __device__(float in, int n) { return in * in; }, Sum<float>(),
+    [=] __device__(float in) { return sqrt(in); });
+
+  thrust::device_ptr<float> d_ptr = thrust::device_pointer_cast(out_labels);
+  float max =
+    *(thrust::max_element(thrust::cuda::par.on(stream), d_ptr, d_ptr + n_rows));
+
+  MLCommon::LinAlg::unaryOp<float>(
+    out_labels, out_labels, n_rows,
+    [=] __device__(float input) { return input / max; }, stream);
+}
+
+class KNNRegressionTest : public ::testing::TestWithParam<KNNRegressionInputs> {
+ protected:
+  void basicTest() {
+    std::shared_ptr<MLCommon::deviceAllocator> alloc(
+      new defaultDeviceAllocator);
+    cudaStream_t stream;
+    cudaStreamCreate(&stream);
+
+    cublasHandle_t cublas_handle;
+    cublasCreate(&cublas_handle);
+
+    cusolverDnHandle_t cusolverDn_handle;
+    cusolverDnCreate(&cusolverDn_handle);
+
+    params = ::testing::TestWithParam<KNNRegressionInputs>::GetParam();
+
+    allocate(train_samples, params.rows * params.cols);
+    allocate(train_labels, params.rows);
+
+    allocate(pred_labels, params.rows);
+
+    allocate(knn_indices, params.rows * params.k);
+    allocate(knn_dists, params.rows * params.k);
+
+    generate_data(train_samples, train_labels, params.rows, params.cols,
+                  stream);
+
+    float **ptrs = new float *[1];
+    int *sizes = new int[1];
```

Comment:
`delete` is missing.

Reply:
I'll update this tests to use vectors as well. Not a fan of the double pointers.